TelemetryManager/Src/TMCircularBuffer.cpp

```diff
@@ -1,30 +1,56 @@
 
-#include "CircularBuffer.hpp"
+#include "TMCircularBuffer.hpp"
 
-TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(buf, size) {
+TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(*buf, size) {
     // Constructor
 }
 
-TMCircularBuffer::TMCircularBuffer(CircularBuffer buf): CircularBuffer(buf) {
+TMCircularBuffer::TMCircularBuffer(CircularBuffer* buf): CircularBuffer(*buf) {
     // Constructor
 }
 
 TMCircularBuffer::~TMCircularBuffer() {
     // Destructor
-}
+    delete[] buf;
+    buf = nullptr;
+ }
+
+TMCircularBuffer::bool TMCircularBuffer::dequeue(MAVLinkByte &byte) {
```

Comment:
One thing you could do to manipulate/retrieve info from the buffer without needing to access the private variable, is to use the already-made functions in the base class. For example, your enqueue() function contents could be something like 
{
read(youre byte, 1);
}

Since the base class already has most circular buffer methods implemented, to save yourself work, you can just use them.

Also, dequeue should return the dequeued byte so we can use it, not a boolean.